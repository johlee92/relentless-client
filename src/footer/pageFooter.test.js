import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import PageFooter from './pageFooter';
import { BrowserRouter } from 'react-router-dom';

describe(`PageFooter component`, () => {
    const props = {
        className: 'test-class-name'
    }

    it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(
            <BrowserRouter> 
                <PageFooter />
            </BrowserRouter>, div);
        ReactDOM.unmountComponentAtNode(div);
    });

    it('renders the component by default', () => {
        const wrapper = shallow(<PageFooter />)
        expect(toJson(wrapper)).toMatchSnapshot()
    })

    it('renders the component from props', () => {
        const wrapper = shallow(<PageFooter {...props} />)
        expect(toJson(wrapper)).toMatchSnapshot()
    })
})
