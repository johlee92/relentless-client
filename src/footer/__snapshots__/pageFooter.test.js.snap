// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageFooter component renders the component by default 1`] = `
<footer
  className="PageFooter"
>
  <section>
    <img
      alt="creator logo"
      className="content-info"
      src="logo.png"
    />
  </section>
  <section>
    Project by JYL
  </section>
  <section>
    <a
      href="mailto:johnathanlee72@hotmail.com"
    >
      <img
        alt="JYL email"
        className="content-info email"
        src="mail.jpg"
      />
    </a>
    <a
      href="https://linkedin.com"
      rel="noopener noreferrer"
      target="_blank"
    >
      <img
        alt="JYL linkedin"
        className="content-info linkedin"
        src="linkedin.jpg"
      />
    </a>
    <a
      href="https://github.com/johlee92"
      rel="noopener noreferrer"
      target="_blank"
    >
      <img
        alt="JYL GitHub"
        className="content-info github"
        src="GitHub.png"
      />
    </a>
  </section>
</footer>
`;

exports[`PageFooter component renders the component from props 1`] = `
<footer
  className="PageFooter"
>
  <section>
    <img
      alt="creator logo"
      className="content-info"
      src="logo.png"
    />
  </section>
  <section>
    Project by JYL
  </section>
  <section>
    <a
      href="mailto:johnathanlee72@hotmail.com"
    >
      <img
        alt="JYL email"
        className="content-info email"
        src="mail.jpg"
      />
    </a>
    <a
      href="https://linkedin.com"
      rel="noopener noreferrer"
      target="_blank"
    >
      <img
        alt="JYL linkedin"
        className="content-info linkedin"
        src="linkedin.jpg"
      />
    </a>
    <a
      href="https://github.com/johlee92"
      rel="noopener noreferrer"
      target="_blank"
    >
      <img
        alt="JYL GitHub"
        className="content-info github"
        src="GitHub.png"
      />
    </a>
  </section>
</footer>
`;
